native/packages/react-native-bpk-component-card/src/BpkCard.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  View,
+  Platform,
+  StyleSheet,
+  TouchableHighlight,
+} from 'react-native';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const tokens = Platform.select({
+  ios: () => require('bpk-tokens/tokens/ios/base.react.native.common.js'), // eslint-disable-line global-require
+  android: () => require('bpk-tokens/tokens/android/base.react.native.common.js'), // eslint-disable-line global-require
+})();
+
+const {
+  colorGray50,
+  colorGray900,
+  colorWhite,
+  spacingBase,
+} = tokens;
+
+/**
+ * Define styles needed by the component
+ */
+const styles = StyleSheet.create({
+  card: {
+    // TODO: update once tokens are available
+    backgroundColor: colorWhite,
+    elevation: 3,
+    shadowColor: colorGray900,
+    shadowOffset: { width: 0, height: 1 },
+    shadowOpacity: 0.3,
+    shadowRadius: 3,
+  },
+  common: {
+    // TODO: update once tokens are available
+    borderRadius: 8,
+  },
+  overflowHidden: {
+    // TODO: update once tokens are available
+    overflow: 'hidden',
+  },
+  padded: {
+    padding: spacingBase,
+  },
+});
+
+const BpkCard = (props) => {
+  const { padded, children, onPress, accessibilityLabel, ...rest } = props;
+
+  const cardStyle = [styles.card, styles.common];
+
+  if (padded) {
+    cardStyle.push(styles.padded);
+  }
+
+  return (
+
+    <TouchableHighlight
+      accessibilityComponentType="button"
+      accessibilityLabel={accessibilityLabel}
+      onPress={onPress}
+      style={cardStyle}
+      underlayColor={colorGray50}
+      {...rest}
+    >
+      <View
+        style={[styles.overflowHidden, styles.common]}
+      >
+        {children}
+      </View>
+    </TouchableHighlight>
+  );
+};
+
+BpkCard.propTypes = {
+  accessibilityLabel: PropTypes.string,
+  children: PropTypes.node,
+  onPress: PropTypes.func.isRequired,
```

Comment:
I can see your motivation for making `onPress` required, by in my experience it's better to be less opinionated about what is strictly required - many times we've had to make breaking changes to relax required props down the line because there has been a valid use case that we haven't thought of.

The web counterpart doesn't have `onClick` required for instance (see http://backpack.prod.aws.skyscnr.com/components/web/cards/).

Happy to get more input on this too btw.

Reply:
Should `BpkButton` mirror this too?

Reply:
I agree with the not making the `onPress` mandatory, off the back of a discussion from yesterday the end result was "a card should be clickable so onPress is required", that leaves to the point tough if there is no onPress should a card use TouchHighlight? it'll give the pressed feedback when nothing it's actually happening

Reply:
After a brief discussion, we came to an agreement to leave the onPress required, for now, it'll be easy to change in the future and making it nor required will not be a breaking change